src/components/GoalsComponents/GoalSublistPage/GoalSublistPage.tsx

```diff
@@ -51,25 +59,21 @@ export const GoalSublist = () => {
   };
   useEffect(() => {
     getGoal(goalID).then((parent) => setParentGoal(parent));
-    setTapCount([-1, 0]);
+    setTapCount({open: -1, click: 0});
   }, [goalID]);
 
   useEffect(() => {
     getChildrenGoals(goalID)
       .then((fetchedGoals) => {
         setChildrenGoals(fetchedGoals);
-        setTapCount([-1, 0]);
+        setTapCount({open: -1, click: 0});
       });
   }, [parentGoal, showAddGoal, showSuggestionModal, showUpdateGoal]);
 
-  const archiveMyGoal = async (goal: GoalItem) => {
-    await archiveUserGoal(goal);
-    await getChildrenGoals(goalID).then((fetchedGoals) => setChildrenGoals(fetchedGoals));
-  };
   const removeChildrenGoal = async (goalId: string) => {
     if (parentGoal?.sublist) {
       // delete subgoals of this goal
-      removeChildrenGoals(goalId);
+      await removeChildrenGoals(goalId);
```

Comment:
we don't need to wait for this. as we are deleting the root goal in the next line, the sublist will be unreachable anyhow. so please do not use await here here